lib/compiler-llvm/src/object_file.rs

```diff
@@ -212,75 +158,86 @@ where
     }
 
     while let Some(section_index) = worklist.pop() {
-        for reloc in reloc_sections
-            .get(&section_index)
-            .iter()
-            .flat_map(|inner| inner.iter().flat_map(|inner2| inner2.iter()))
+        for (offset, reloc) in elf
+            .section_by_index(section_index)
+            .map_err(map_object_err)?
+            .relocations()
         {
-            let kind = match reloc.r_type {
-                // TODO: these constants are not per-arch, we'll need to
-                // make the whole match per-arch.
-                goblin::elf::reloc::R_X86_64_64 => RelocationKind::Abs8,
-                goblin::elf::reloc::R_X86_64_PC64 => RelocationKind::X86PCRel8,
-                goblin::elf::reloc::R_X86_64_GOT64 => {
-                    return Err(CompileError::Codegen(
-                        "unimplemented PIC relocation R_X86_64_GOT64".into(),
-                    ));
-                }
-                goblin::elf::reloc::R_X86_64_GOTPC64 => {
-                    return Err(CompileError::Codegen(
-                        "unimplemented PIC relocation R_X86_64_GOTPC64".into(),
-                    ));
-                }
+            let kind = match (reloc.kind(), reloc.size()) {
+                (object::RelocationKind::Absolute, 64) => RelocationKind::Abs8,
+                (object::RelocationKind::Elf(24), 0) => RelocationKind::X86PCRel8,
```

Comment:
Can we have here the proper static variable here instead of `24`? (I think the `object` has defined most of those)

Reply:
Done.